src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -968,7 +968,7 @@ class npc_scarlet_miner_cart : public CreatureScript
         {
             me->SetImmuneToAll(true);
             me->SetFaction(FACTION_FRIENDLY);
-            me->SetDisplayId(me->GetCreatureTemplate()->Modelid1); // Modelid2 is a horse.
+            me->SetDisplayFromModel(0); // Modelid2 is a horse.
```

Comment:
are these right?

Reply:
Might be a part of the objective of removing past hacks to step over the probability problem.

Reply:
Just checked brute data, it's definitely a probability hack.